spec/datadog/tracing/contrib/utils/quantization/http_spec.rb

```diff
@@ -393,14 +405,28 @@
           it { is_expected.to eq('json=%7B%20<redacted>%7D') }
         end
 
+        context 'with a freestanding value match' do
+          let(:query) { 'k=ssh-rsa%20AAAAB3NzaC1yc2EAAAADAQABAAABgQCxkfwLCG9IW32wN+dbys1JP1qRTA9QRn8XnZdr5Irz7yGXCkPjr7e51F/v73fRk4FOnaRjHartimlsb4NogaNN1clkHkvrRvjHyTzGBM/WgumEdco0Ay4H/BdyJwLNw88iUGVM3H93s+rP5a8u4Ptk4HZy1lSWAjg52ZhYv3V+cyQNjfHiOuHk2lChmtTE1FlKfc8pkxB4QnyG3EbQrEDzRkym8NTlvZIaOv01VdsmRXWurigI9AnqgoA43eD3JqWswjyCkmSnnW3h+9iB4bgRrSIT2QBugpIHVM4pbkT4UrlpcVEeAziYNPC4H61luX0d1/qOntaMFMtvJKuG/3Jj8t19O+nSTa9re+PzS5Qm8J/7xzaxxOIq94D7FnGPG5JifJ4WZRWF0PDptOXbJY72Yumb+xaJtr9LlqfS6fJzWkA68PT/ZaEk5JgaPHhlvUvmx5JlmP7u2eFcbNHpHQHoIfJjXk8J3EDXjl5Rp6E9KxARD3p45Sh8b4LoVR8Z0+k=%20dummy' } # rubocop:disable Layout/LineLength
+          let(:options) { { obfuscate: :internal } }
+
+          it { is_expected.to eq('k=<redacted>') }
+        end
+
+        context 'with a freestanding value in URL-encoded JSON' do
+          let(:query) { 'json=%7B%20%22k%22%3A%20%22ssh-rsa%20AAAAB3NzaC1yc2EAAAADAQABAAABgQCxkfwLCG9IW32wN%2Bdbys1JP1qRTA9QRn8XnZdr5Irz7yGXCkPjr7e51F%2Fv73fRk4FOnaRjHartimlsb4NogaNN1clkHkvrRvjHyTzGBM%2FWgumEdco0Ay4H%2FBdyJwLNw88iUGVM3H93s%2BrP5a8u4Ptk4HZy1lSWAjg52ZhYv3V%2BcyQNjfHiOuHk2lChmtTE1FlKfc8pkxB4QnyG3EbQrEDzRkym8NTlvZIaOv01VdsmRXWurigI9AnqgoA43eD3JqWswjyCkmSnnW3h%2B9iB4bgRrSIT2QBugpIHVM4pbkT4UrlpcVEeAziYNPC4H61luX0d1%2FqOntaMFMtvJKuG%2F3Jj8t19O%2BnSTa9re%2BPzS5Qm8J%2F7xzaxxOIq94D7FnGPG5JifJ4WZRWF0PDptOXbJY72Yumb%2BxaJtr9LlqfS6fJzWkA68PT%2FZaEk5JgaPHhlvUvmx5JlmP7u2eFcbNHpHQHoIfJjXk8J3EDXjl5Rp6E9KxARD3p45Sh8b4LoVR8Z0%2Bk%3D%20dummy%22%7D' } # rubocop:disable Layout/LineLength
+          let(:options) { { obfuscate: :internal } }
+
+          it { is_expected.to eq('json=%7B%20%22k%22%3A%20%22<redacted>%22%7D') }
```

Comment:
I guess in that case it should be one of these:

```
# invalid JSON, but it's meant to be displayed only anyway, not parsed
it { is_expected.to eq('json=%7B%20%22<redacted>%22%7D') }

# valid JSON, it would be more complex to implement and not sure it is better since multiple redactions would have multiple <redacted> keys anyway
it { is_expected.to eq('json=%7B%20%22<redacted>%22%3A%20%22<redacted>%22%7D') }
```

The regex would get more complicated (accepting any json key) but I could give it a try.

Reply:
Do we decode the URL to redact it? What kind of magic makes 'json=%7B%20%22k%22%3A%20%22<redacted>%22%7D' work like this?